plugin/npm-module/monitor.js

```diff
@@ -10,7 +10,10 @@ module.exports = class MonitorStats {
   constructor(options) {
     this.options = Object.assign({
```

Comment:
is it possible to deep merge the options with the default to prevent overriding of the nested properties which the user doesn't meant to change? e.g `new WebpackMonitor({ jsonOpts: { version: false } })` would also override the `jsonOpts.source: false` property.